src/OpenTelemetry.Exporter.Geneva/Internal/MsgPack/MsgPackTraceExporter.cs

```diff
@@ -367,12 +458,38 @@ internal ArraySegment<byte> SerializeActivity(Activity activity)
         var isStatusSuccess = true;
         string? statusDescription = null;
 
+        var isServerActivity = activity.Kind == ActivityKind.Server;
+        var httpUrlParts = this.HttpUrlParts.Value ?? new object?[CS40_PART_B_HTTPURL_MAPPING.Count];
+        if (isServerActivity)
+        {
+            if (this.HttpUrlParts.Value == null)
+            {
+                this.HttpUrlParts.Value = httpUrlParts;
+            }
+            else
+            {
+                Array.Clear(httpUrlParts, 0, httpUrlParts.Length);
+            }
+        }
+
+        string? httpMethodString = null;  // Used to determine which HTTP version to use for httpUrl.
+
         foreach (ref readonly var entry in activity.EnumerateTagObjects())
         {
+            if (isServerActivity && CacheIfPartOfHttpUrl(entry, httpUrlParts))
+            {
+                continue; // Skip this entry, since it is part of httpUrl.
+            }
+
             // TODO: check name collision
             if (CS40_PART_B_MAPPING.TryGetValue(entry.Key, out var replacementKey))
             {
                 cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, replacementKey);
+
+                if (isServerActivity && replacementKey == "httpMethod")
```

Comment:
Remove the logic to check httpMethod from here.

Reply:
Updated.